jgt_tools/docs/build_docs.py

```diff
@@ -95,9 +92,16 @@ def build():
         shutil.rmtree(str(BASE_DIR / DOCS_OUTPUT_DIRECTORY), ignore_errors=True)
         shutil.rmtree(str(BASE_DIR / DOCS_WORKING_DIRECTORY), ignore_errors=True)
 
-    _build_conf()
+    pyproject = get_pyproject_config()
+    poetry = pyproject["tool"]["poetry"]
+    dependencies = list(
+        itertools.chain(poetry["dependencies"], poetry["dev-dependencies"])
+    )
+
+    if default_commands("build_docs") or "sphinx" in dependencies:
```

Comment:
Must be post-prandial bran fog, but I keep staring at this going "shoudn't that `or` be `and`" ?

Reply:
If the user is using the default commands, those need sphinx and the conf file. If the user is **not** using the default commands, then we check to see if "sphinx" has been added as a dependency, in which case we go ahead and build the conf file.

Reply:
However, I don't think this check will work as expected in some cases, because if you install `sphinx-autobuild`, poetry will not add `sphinx` to the `pyproject.toml` file (since it's a secondary dependency, and those are tracked in `poetry.lock`), and I'm not sure I want to go down the route of seeing if the word "sphinx" is in every dependency...

Reply:
Hmmm, good point. So what would be a better solution?  Just accepting this as marginally better?

Reply:
@dgou I'm going to ruminate on it over the weekend and see if I find "inspiration"; otherwise I'll probably close this PR because I'm concerned there _could_ be an edge case that bites a user that they don't understand why it doesn't behave as expected.

Reply:
Meant to get to this Monday, but got distracted. Rather than checking `dependencies`, we could just check if the library is installed like `if pkgutil.find_module("sphinx")` and that would even bypass the need for the `default_commands` check. _**Thoughts?**_

Reply:
Hmm, if you mean: https://docs.python.org/3.7/library/pkgutil.html#pkgutil.find_loader
Then maybe use https://docs.python.org/3.7/library/importlib.html#checking-if-a-module-can-be-imported
I haven't fully reloaded the context for this into my brain though, so I might be off kilter.

Reply:
One is a wrapper around the other, so yes, either works. It's just a quick way for us to determine if Sphinx is installed before making a Sphinx config file.

Reply:
Yup, so I'm fine with that since it should work without having to actually load anything.
I'd say go for it.